### DISCUSSION TITLE
Setting static Cache-Control headers for Next image component

### DISCUSSION THREAD

**Discussion by KevinFerm** [vercel/next.js#20389] [Help]
I would like to set a static Cache-Control header for all Next image components, both for internal and external images.
I'm using Vercel.json currently, so I've tried with the vercel.json eg.
```json
{
  "version": 2,
  "headers": [
    {
      "source": "/_next/image/(.*)",
      "headers": [{ "key": "Cache-Control", "value": "s-maxage=604800" }]
    }
  ]
}
```
However, this doesn't work, as it appears to ignore the header, no matter what I do.

Is there a way to do this with next.js configuration instead?

**Comment by badsyntax** [comment]
I'm also struggling with this. IMO the instructions aren't clear. What is the upstream server in this context? Is it the nextjs server itself? 

Assuming the nextjs server is the "upstream" server, I set the cache-control headers on the "source" image like so:

next.config.js:

```js
module.exports = {
  async headers() {
    return [
      {
        source: '/images/(.*)',
        headers: [
          {
            key: 'Cache-Control',
            value: 'public, max-age=31536000, immutable',
          },
        ],
      },
    ]
  },
}
```

Now when viewing the raw source image, the cache-control headers are set correctly, but when trying to view an optimised image (via _next/image), the cache controls headers are NOT set.

I don't understand why the docs don't cover the most basic caching scenario which covers the example listed on the docs page. 

Possibly related: https://github.com/vercel/next.js/issues/19914

**Comment by leerob** [comment]
Does this configuration option work for you?

https://nextjs.org/docs/app/api-reference/components/image#minimumcachettl
